src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java

```diff
@@ -278,7 +278,8 @@ public void databaseClosing() {
     protected void close() {
         // shutdown ScheduledExecutorService. Without shutdown, cause thread leak
         if (remoteRequestExecutor != null && !remoteRequestExecutor.isShutdown()) {
-            Utils.shutdownAndAwaitTermination(remoteRequestExecutor);
+            // Note: Time to wait is set 60 sec because RemoteRequest's socket timeout is set 60 seconds.
+            Utils.shutdownAndAwaitTermination(remoteRequestExecutor, 60);
         }
```

Comment:
Solution to abort connection.

- `HttpClient.getConnectionManager().shutdown();` might make connection aborted
http://stackoverflow.com/questions/7140565/how-do-i-cancel-an-android-http-request

- https://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/client/HttpClient.html#getConnectionManager()
- https://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/conn/ClientConnectionManager.html#shutdown()